client/process.go

```diff
@@ -84,8 +84,15 @@ func (config *Config) RunProcess(commandLine string, stdin io.Reader, stdout io.
 		}
 
 		// Don't need stdin now we've sent everything. This signals GCS that we are finished sending data.
-		if err := process.Process.CloseStdin(); err != nil {
-			return nil, err
+		err := process.Process.CloseStdin()
+		if err != nil {
+			if !hcsshim.IsNotExist(err) && !hcsshim.IsAlreadyClosed(err) {
+				// This error will occur if the compute system is currently shutting down
+				if perr, ok := err.(*hcsshim.ProcessError); ok && perr.Err != hcsshim.ErrVmcomputeOperationInvalidState {
```

Comment:
Not for this PR, but we should add this to one of the error checking functions in hcsshim rather than inspecting the errors in consumers.

Reply:
I'm not sure, there's that example in tar2vhd where I think ANY error is legitimate. But, yes, it could be refactored still later.

Reply:
I mean moving this check into hcsshim.IsAlreadyClosed, as this error is just a symptom of the process already being closed. That way we can ignore it here, and still error in tar2vhd